File: backend/app/schemas/outreach.py
from pydantic import BaseModel


class CreateOutreachRequest(BaseModel):
    person_id: str
    job_id: str | None = None
    message_id: str | None = None
    status: str = "draft"
    channel: str | None = None
    notes: str | None = None
    last_contacted_at: str | None = None
    next_follow_up_at: str | None = None


class UpdateOutreachRequest(BaseModel):
    status: str | None = None
    channel: str | None = None
    notes: str | None = None
    job_id: str | None = None
    message_id: str | None = None
    last_contacted_at: str | None = None
    next_follow_up_at: str | None = None
    response_received: bool | None = None


class OutreachResponse(BaseModel):
    id: str
    person_id: str
    job_id: str | None
    message_id: str | None
    status: str
    channel: str | None
    notes: str | None
    last_contacted_at: str | None
    next_follow_up_at: str | None
    response_received: bool
    person_name: str | None = None
    person_title: str | None = None
    company_name: str | None = None
    job_title: str | None = None
    created_at: str
    updated_at: str

    model_config = {"from_attributes": True}


class OutreachStats(BaseModel):
    total_contacts: int
    by_status: dict[str, int]
    response_rate: float
    upcoming_follow_ups: int

File: frontend/src/types/outreach.ts
// Outreach Tracker types
export type OutreachStatus = 'draft' | 'sent' | 'connected' | 'responded' | 'met' | 'following_up' | 'closed';

export type OutreachChannel = 'linkedin_note' | 'linkedin_message' | 'email' | 'phone' | 'in_person' | 'other';

export interface OutreachLog {
  id: string;
  person_id: string;
  job_id: string | null;
  message_id: string | null;
  status: OutreachStatus;
  channel: OutreachChannel | null;
  notes: string | null;
  last_contacted_at: string | null;
  next_follow_up_at: string | null;
  response_received: boolean;
  person_name: string | null;
  person_title: string | null;
  company_name: string | null;
  job_title: string | null;
  created_at: string;
  updated_at: string;
}

export interface OutreachStats {
  total_contacts: number;
  by_status: Record<string, number>;
  response_rate: number;
  upcoming_follow_ups: number;
}

export interface CreateOutreachRequest {
  person_id: string;
  job_id?: string;
  message_id?: string;
  status?: OutreachStatus;
  channel?: OutreachChannel;
  notes?: string;
  last_contacted_at?: string;
  next_follow_up_at?: string;
}

export interface UpdateOutreachRequest {
  status?: OutreachStatus;
  channel?: OutreachChannel;
  notes?: string;
  job_id?: string;
  message_id?: string;
  last_contacted_at?: string;
  next_follow_up_at?: string;
  response_received?: boolean;
}
